main.py

```diff
@@ -146,23 +58,64 @@ def new_task():
             return_to_menu()
 
 
+def task_edition():
+    task_number = input("Podaj numer ID zadania do edycji: ")
+    if task_number.isdecimal() is False:
+        print("Błędny wpis numeru zadania!")
+        task_edition()
+    elif task_number.isdecimal() is True:
```

Comment:
Gdy sprawdzasz bool, to są tylko 2 opcje, więc wystarczy
```python
if ... == True
    ...
else
    <kod w przeciwnym wypadku>
```
Albo można jeszcze inaczej
```python
if ... == True:
    ...
<kod w przeciwnym wypadku>
```
Z tym że tutaj drugi sposób nie przejdzie bo jak potem wrócisz z nowego task_edition (z linii 65) to wykona się kod po if. Tutaj jest trochę skopane ze sposobem w jaki jest wywoływany task edition w przypadku błędu. Musiałbym to przekminić

Reply:
Funkcjonalnie z task_edition to jestem bardzo zadowolony :D Pełna edycja plus zabezpieczenia przed głupotami użytkownika, jest super. No ale wygląd kodu się trochę zagmatwał :D

Reply:
Ok sprawdziłem, działa 🗡️ 
``` python
     if task_number.isdecimal() is False:
        print("Błędny wpis numeru zadania!")
        task_edition()
    else:

Reply:
Dla lepszego wyglądu powinienem to odwrócić:
```python
def task_edition():
    task_number = input("Podaj numer ID zadania do edycji: ")
    if task_number.isdecimal() is True:
        task_number = int(task_number)
        with conn:
            sel_task = select_task_by_id(conn, task_number)
            if sel_task is not None:
                choice = input("Edycja Statusu, nazwy czy usunięcie zadania [S/n/u]? ")
                if choice.lower() == "s" or len(choice) == 0:
                    sel_task_status = bool(sel_task[2])
                    choice = input(f"Potwierdzasz zmianę statusu zadania na {not sel_task_status} (T)? ")
                    if choice.lower() == "t" or len(choice) == 0:
                        update_task_status(conn, (not sel_task_status, task_number))
                    else:
                        main_menu()
                elif choice.lower() == "n":
                    print(sel_task[1])
                    update_name = input("Podaj poprawną nazwę edytowanego zadania: ")
                    update_task_name(conn, (update_name, task_number))
                    print("Nazwę zadania pomyślnie zmieniono w bazie danych")
                elif choice.lower() == "u":
                    delete_task(conn, task_number)
                    print("Zadanie pomyślnie usunięto z bazy danych")
                else:
                    print("Błędny wybór")
                    return_to_menu()
            else:
                print("Brak wybranego numeru zadania w tabeli bazy danych")
                select_all_tasks(conn)
                task_edition()
    else:
        print("Błędny wpis numeru zadania!")
        task_edition()